keras/ops/image.py

```diff
@@ -501,3 +501,157 @@ def map_coordinates(
         fill_mode,
         fill_value,
     )
+
+
+class PadToBoundingBox(Operation):
+    def __init__(
+        self,
+        offset_height,
+        offset_width,
+        target_height,
+        target_width,
+        check_dims=True,
+    ):
+        super().__init__()
+        self.offset_height = offset_height
+        self.offset_width = offset_width
+        self.target_height = target_height
+        self.target_width = target_width
+        self.check_dims = check_dims
+
+    def call(self, image):
+        return _pad_to_bounding_box(
+            image,
+            self.offset_height,
+            self.offset_width,
+            self.target_height,
+            self.target_width,
+            self.check_dims,
+        )
+
+    def compute_output_spec(self, image):
+        out_shape = (
+            image.shape[0],
+            self.target_height,
+            self.target_width,
+            image.shape[-1],
+        )
+        if len(image.shape) == 3:
+            out_shape = out_shape[1:]
+        return KerasTensor(
+            shape=out_shape,
+            dtype=image.dtype,
+        )
+
+
+def _pad_to_bounding_box(
+    image, offset_height, offset_width, target_height, target_width, check_dims
+):
+    image = backend.convert_to_tensor(image)
+    is_batch = True
+    image_shape = image.shape
+    if len(image_shape) == 3:
+        is_batch = False
+        image = backend.numpy.expand_dims(image, 0)
+    elif len(image_shape) != 4:
+        raise ValueError(
+            "'image' (shape %s) must have either 3 or 4 dimensions."
+            % image_shape
+        )
+
+    batch, height, width, depth = image.shape
+    after_padding_width = target_width - offset_width - width
+    after_padding_height = target_height - offset_height - height
+
+    if check_dims:
+        if not offset_height >= 0:
+            raise ValueError("offset_height must be >= 0")
+        if not offset_width >= 0:
+            raise ValueError("offset_width must be >= 0")
+        if not after_padding_width >= 0:
+            raise ValueError("width must be <= target - offset")
+        if not after_padding_height >= 0:
+            raise ValueError("height must be <= target - offset")
+
+    paddings = backend.numpy.reshape(
+        backend.numpy.stack(
+            [
+                0,
+                0,
+                offset_height,
+                after_padding_height,
+                offset_width,
+                after_padding_width,
+                0,
+                0,
+            ]
+        ),
+        [4, 2],
+    )
+    padded = backend.numpy.pad(image, paddings)
+
+    padded_shape = [batch, target_height, target_width, depth]
+    padded = backend.numpy.reshape(padded, padded_shape)
+
+    if not is_batch:
+        padded = backend.numpy.squeeze(padded, axis=[0])
+    return padded
+
+
+@keras_export("keras.ops.image.pad_to_bounding_box")
+def pad_to_bounding_box(
```

Comment:
The API is a bit strange to me -- it says "pad to bounding box" but there's no bounding box being specified.

Reply:
Yes I agree, should we consider a different api??

Reply:
I think so -- what could the API look like? `pad_image`?

Also, it is optimal to ask the user to specify `offset_height, offset_width, target_height, target_width`? It could be something else, e.g. `right_padding, left_padding, top_padding, bottom_padding`. Which may be clearer.

Reply:
Yes `pad_image` will be nice. I think having `target_width` and `target_height` would be nice in a sense that in a lot of cases we know the final output shape so we won't have to calculate the padding sizes manually.

Reply:
Thanks for the input! Do you think it's better to stay with the TF signature, or expand it with something like

`right_padding=None, left_padding=None, top_padding=None, bottom_padding=None, target_width=None, target_height=None` -- you could specify e.g. `right_padding=..., target_width=...` or `right_padding=..., bottom_padding=...`.

Reply:
Apologies for the delay once again; I've just returned home after attending the conference abroad.

I believe expanding the functionality would be a valuable addition for our future work. As a friendly reminder, we need to apply the same changes to the `crop_to_bounding_box` implementation since it uses the same signature:

```py
tf.image.crop_to_bounding_box(
    image, offset_height, offset_width, target_height, target_width
)
```

I will submit a PR for `crop` after this one.

Reply:
@fchollet, since `pad_to_bounding_box` and `crop_to_bounding_box` are related functions and we're considering a change in the signature for `pad` from using `offset_height/width` to `left/right/top/bottom_padding`. Should we make a corresponding change for `crop` to use `left/right/top/bottom_cropping` for consistency?

Reply:
Yes, sure. It should likely be called `pad_images` and `crop_images` to distinguish from `ops.pad`.

Reply:
Thanks for the feedback, I start adding the features you mentioned.